scripts/jenkins/qa_devstack.sh

```diff
@@ -58,8 +58,10 @@ function h_setup_base_repos {
     if [[ $DIST_NAME == "openSUSE_Leap" ]]; then
         $zypper ar -f http://download.opensuse.org/distribution/leap/${DIST_VERSION}/repo/oss/ Base || true
         $zypper ar -f http://download.opensuse.org/update/leap/${DIST_VERSION}/oss/openSUSE:Leap:${DIST_VERSION}:Update.repo || true
-        # Python 3.x support is quite broken atm
-        if false && [[ $DIST_VERSION == "15.0" ]]; then
+        # Python 2.7 is scheduled for end-of-life in 2020
+        # Openstack goal is to have python3 supported on the end of the T cycle
```

Comment:
We're not at the end of the T cycle yet, does everything actually work with python3? Swift has been the main holdout.

Reply:
well, that's the goal, isn't it? to find the errors and fix them
IMHO this jobs add more value if we can catch upstream problems soon

Reply:
I think, the plan is to drop python-2.7 in SLE-15-SP1, so it will not stay around for long and we better put in effort to get everything working with 3.x soon.

Reply:
@cmurphy also without this patch this job is going to be red always https://github.com/SUSE-Cloud/automation/pull/2809#issuecomment-425343893

Reply:
If this change makes the job green that's great - you checked, right? My only concern is making things worse if swift is still known not to work with python3.

Reply:
seems that somebody added a fix for the swift problem 
https://github.com/SUSE-Cloud/automation/blob/de7661951cbf88e6b231b8c5a76dd7896b2e8d3a/scripts/jenkins/qa_devstack.sh#L136

Reply:
Hmm well if we always use python3 then we'll never be testing swift which is not great

Reply:
I hate these situations, another solution is to disable tls, to avoid the bug with python2 and urllib3 using ipv6 addresses inside the DNS x509 SAN fields, but I'm sure that sooner or later it's going to backfire because we are hacking a hack to avoid a known issue that's not going to be fixed :)

Reply:
> we'll never be testing swift

We will, just not with devstack on the distribution that will drop support for python2.7. The rpm packages of swift will still run through the openstack-cleanvm job on Leap 15. We could add a job to test the previous Leap release with devstack. Or fix swift for python3 support.